test/system_test.py

```diff
@@ -74,23 +80,28 @@ def spack_install_and_test(spec: str,
     log_with_spack(f'spack spec {spec}',
                    'system_test',
                    log_filename,
-                   srun=False)
+                   srun=False,
+                   uenv=uenv)
     if split_phases:
         log_with_spack(
             f'spack {command} --until build --test=root -n -v {spec}',
             'system_test',
             log_filename,
-            srun=True)
+            srun=not machine_name() == 'balfrin',
```

Comment:
The idea of deactivating srun on balfrin used to be in exactly 1 line of code. Now its spread out. Clearly a degradation of the quality of this code.

Reply:
BTW: `not A == B` is equivalent to `A != B` and often considered simpler 😉 
[Prove by George Boole (1847)](https://www.gutenberg.org/ebooks/36884)

Reply:
True that

Reply:
restored the brackets